vllm/v1/structured_output/__init__.py

```diff
@@ -7,104 +7,58 @@
 
 from vllm.config import VllmConfig
 from vllm.logger import init_logger
-from vllm.transformers_utils.tokenizer_group import init_tokenizer_from_configs
-from vllm.transformers_utils.tokenizers.mistral import MistralTokenizer
-from vllm.utils import LazyLoader
-from vllm.v1.structured_output.grammar import Grammar, StructuredOutputOptions
+from vllm.v1.structured_output.backend_types import (StructuredOutputBackend,
+                                                     StructuredOutputGrammar)
+from vllm.v1.structured_output.backend_xgrammar import XgrammarBackend
 
 if TYPE_CHECKING:
     import numpy as np
     import numpy.typing as npt
-    import xgrammar as xgr
+    import torch
 
     from vllm.v1.request import Request
-else:
-    xgr = LazyLoader("xgr", globals(), "xgrammar")
 
 logger = init_logger(__name__)
 
 
 class StructuredOutputManager:
+    """Engine-level manager for structured output requests."""
 
     def __init__(self, vllm_config: VllmConfig):
-        self.vocab_size = vllm_config.model_config.get_vocab_size()
+        self.backend: Optional[StructuredOutputBackend] = None
```

Comment:
fwiw I don't think backend should be None.

Backend should be server-side, idk if changing backend per request is beneficial atm.

Reply:
Right - changing per request is explicitly blocked.

It's None here to defer initialization until it's needed for the first time.

Reply:
> Right - changing per request is explicitly blocked.

actually I think this part is in my commit adding the next backend, but either way, agree we shouldn't support it